server/app/app.go

```diff
@@ -25,14 +30,14 @@ type App struct {
 	config       *config.Configuration
 	store        store.Store
 	auth         *auth.Auth
-	wsServer     *ws.Server
+	wsServer     WebsocketServer
 	filesBackend filestore.FileBackend
 	webhook      *webhook.Client
 	metrics      *metrics.Metrics
 	logger       *mlog.Logger
 }
 
-func New(config *config.Configuration, wsServer *ws.Server, services Services) *App {
+func New(config *config.Configuration, wsServer WebsocketServer, services Services) *App {
```

Comment:
Is there any reason to not follow the proposed approach of `type App struct {services Services}`, also is there any reason to pass the `wsServer` outside and inside the `services` param? (I know that are not related to your changes).

Reply:
We end up with calls like `a.services.logger.Debug(...)` which gets tedious. I think we still achieve the abstraction/separation even though the App has each service as a separate field.

The only reason the wsServer` is not passed in the services struct is because `ws.Server` is not in the services package.

Reply:
ok, makes sense.